Scripts/build.swift

```diff
@@ -115,31 +164,43 @@ guard CommandLine.arguments.count > 1 else {
-try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/WATCHOS_DEPLOYMENT_TARGET = \"7.0\"/MACOSX_DEPLOYMENT_TARGET = \"5.0\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
-try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/MACOSX_DEPLOYMENT_TARGET = \"11.0\"/MACOSX_DEPLOYMENT_TARGET = \"10.15\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
-
 let rawPlatforms = CommandLine.arguments[1].components(separatedBy: ",")
 
+var isFirstRun = true
 for rawPlatform in rawPlatforms {
     guard let platform = Platform(rawValue: rawPlatform) else {
         print("Received unknown platform type \(rawPlatform)")
         print("Possible platform types are: \(Platform.allCases)")
         throw TaskError.code(1)
     }
+
+    if isFirstRun && platform.shouldGenerateXcodeproj {
+        // Generate the xcode project if we need it
+        try execute(commandPath: "/usr/bin/swift", arguments: ["package", "generate-xcodeproj", "--output=generated/"])
+
+        // The generate-xcodeproj command has a bug where the test deployment target is above the minimum deployment target for the project. Fix it with sed.
+        try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/IPHONEOS_DEPLOYMENT_TARGET = \"14.0\"/IPHONEOS_DEPLOYMENT_TARGET = \"12.0\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
+        try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/TVOS_DEPLOYMENT_TARGET = \"14.0\"/TVOS_DEPLOYMENT_TARGET = \"12.0\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
+        try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/WATCHOS_DEPLOYMENT_TARGET = \"7.0\"/MACOSX_DEPLOYMENT_TARGET = \"5.0\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
+        try execute(commandPath: "/usr/bin/sed", arguments: ["-i", "-e", "s/MACOSX_DEPLOYMENT_TARGET = \"11.0\"/MACOSX_DEPLOYMENT_TARGET = \"11.0\"/g", "generated/CacheAdvance.xcodeproj/project.pbxproj"])
+    }
+
     var xcodeBuildArguments = [
-        "-project", "generated/CacheAdvance.xcodeproj",
         "-scheme", "CacheAdvance-Package",
         "-sdk", platform.sdk,
-        "-configuration", "Release",
         "-derivedDataPath", platform.derivedDataPath,
-        "-PBXBuildsContinueAfterErrors=0",
-        "OTHER_CFLAGS='-DGENERATED_XCODE_PROJECT'",
+        "-PBXBuildsContinueAfterErrors=0"
     ]
+    if platform.shouldGenerateXcodeproj {
+        // Point at the generated project
+        xcodeBuildArguments.append("-project")
+        xcodeBuildArguments.append("generated/CacheAdvance.xcodeproj")
+        // Set the configuration to be release – this configuration is not supported when running xcodebuild without a xcodeproj file.
+        xcodeBuildArguments.append("-configuration")
+        xcodeBuildArguments.append("Release")
```

Comment:
This configuration only existing when we have an xcodeproj file sucks. I managed to continue testing in Release in CI by using `swift test  -c release -Xswiftc -enable-testing`. We're still running via `xcodebuild` for test coverage on Xcode 13, but that run is in debug rather than Release.